src/interface.jl

```diff
@@ -51,7 +51,7 @@ function Base.getindex{T<:Integer}(b::Buffer, r::UnitRange{T})
 end
 
 Base.setindex!(b::Buffer, lines::Array, r::CappedRange) =
-    set_line_slice(b, r.start, r.stop, true, true, lines)
+    set_line_slice(b, r.start, r.stop, true, true, convert(Array{ASCIIString, 1}, lines))
```

Comment:
But this won't work if `lines` contains a `UTF8String`. In what situation is this change useful? Whatever argument we support here, I think we should support for any API function that expects a `ArrayOf(String)`. Eventually, I think what we want to to is if the caller supplies a `Vector{T}` for any `T <: ByteString` we just pass it through, but if a `Vector{T}` where T is a proper supertype (`Any`, essentially) is passed, we should dynamically check each element ( i e by `convert(Vector{ByteString}, val)` ).  I might take a look into this in the weekend.